### PR TITLE
Fix broken links discovered in audit

### PR DESCRIPTION
**Pull request by alfarok**
### Task
This PR resolves the broken links outlined in #3136 

### Goals

- [x] Restore as many broken links from #3136 as possible (outlined below)

### Broken Links

Fixed = ✔️
Removed = 🗑️ (verified content was deleted or no longer available) 
Timing Out = ⌛ (potentially still active)
No Action = ❌

**Audit 1**

|Fixed|File|Error|Tag|Url|
|---|---|---|---|---|
|✔️|[learn-keys.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-keys.md)|**404**|`original forum post`|https://forum.web3.foundation/t/account-signatures-and-keys-in-polkadot/70/2|
|✔️|[learn-keys.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-keys.md)|**404**|`Key discovery attack on BIP32-Ed25519`|https://forum.web3.foundation/t/key-recovery-attack-on-bip32-ed25519/44|
|✔️|[learn-keys.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-keys.md)|**404**|`Account signatures and keys in Polkadot`|https://forum.web3.foundation/t/account-signatures-and-keys-in-polkadot/70|
|🗑️|[learn-validator.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-validator.md)|**404**|`Other Useful Links`|https://forum.web3.foundation/t/useful-links-for-validators/20|
|✔️|[learn-phragmen.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-phragmen.md)|**404**|`off-chain worker`|https://docs.substrate.io/v3/concepts/off-chain-workers|
|✔️|[learn-phragmen.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-phragmen.md)|**404**|`W3F Research Page on Sequential Phragmén Method`|https://research.web3.foundation/en/latest/polkadot/NPoS/4.%20Sequential%20Phragm%C3%A9n%E2%80%99s%20method.html|
|✔️|[learn-account-restore.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-account-restore.md)|**404**|`this document`|https://github.com/paritytech/parity-signer/blob/master/docs/tutorials/Recover-Account-Polkadotjs.md|
|✔️|[learn-accounts.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-accounts.md)|**404**|`Rust source`|https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/src/constants.rs#L26|
|✔️|[learn-accounts.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-accounts.md)|**404**|`Utility Scripts`|https://github.com/w3f/utility-scripts|
|✔️|[learn-accounts.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/learn/learn-accounts.md)|**404**|`pubkeyToAddress.js`|https://github.com/w3f/utility-scripts/blob/master/src/misc/pubkeyToAddress.js|
|✔️|[contributors.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/general/contributors.md)|**404**|`quest for developer health`|https://bruno.id/an-endomorphs-journey-to-health-part-2/|
|🗑️|[grants.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/general/grants.md)|**404**|`Darwinia Grants Program`|https://docs.darwinia.network/docs/en/dev-bounty#grant-program|
|✔️|[kusama-community.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/general/kusama/kusama-community.md)|**404**|`Polkadot's Latest Research`|https://research.web3.foundation/en/latest/polkadot/|
|✔️|[maintain-guides-how-to-validate-kusama.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/maintain/kusama/maintain-guides-how-to-validate-kusama.md)|**301**|`hasKey`|https://polkadot.js.org/api/substrate/rpc.html#haskey-publickey-bytes-keytype-text-bool|
|✔️|[maintain-guides-how-to-validate-kusama.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/maintain/kusama/maintain-guides-how-to-validate-kusama.md)|**301**|`hasSessionKeys`|https://polkadot.js.org/api/substrate/rpc.html#hassessionkeys-sessionkeys-bytes-bool|
|🗑️|[maintain-guides-how-to-validate-kusama.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/maintain/kusama/maintain-guides-how-to-validate-kusama.md)|**404**|`Kusama forum`|https://forum.kusama.network/|
|✔️|[build-protocol-info.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/build/build-protocol-info.md)|**404**|`Lock Withdraw Reasons`|https://docs.substrate.io/rustdocs/latest/frame_support/traits/enum.WithdrawReason.html|
|✔️|[build-protocol-info.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/build/build-protocol-info.md)|**404**|`Vesting`|https://docs.substrate.io/rustdocs/latest/pallet_vesting/struct.Vesting.html|
|✔️|[build-tools-index.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/build/build-tools-index.md)|**301**|`@polkadot/keyring`|https://polkadot.js.org/common/keyring/|
|✔️|[build-tools-index.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/build/build-tools-index.md)|**301**|`@polkadot/util`|https://polkadot.js.org/common/util/|
|✔️|[build-tools-index.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/build/build-tools-index.md)|**301**|`@polkadot/util-crypto`|https://polkadot.js.org/common/util-crypto/|
|✔️|[build-tools-index.md](https://github.com/w3f/polkadot-wiki/blob/master//docs/build/build-tools-index.md)|**404**|`Haskell`|https://github.com/airalab/hs-web3/tree/master/src/Codec|

**Audit 2**
|Fixed|File|Error|Tag|Url|
|---|---|---|---|---|
|✔️|[build-storage.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-storage.md)|**404**|``Storage Value``|https://docs.substrate.io/rustdocs/latest/frame_support/storage/trait.StorageValue.html|
|✔️|[build-storage.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-storage.md)|**404**|``Storage Map``|https://docs.substrate.io/rustdocs/latest/frame_support/storage/trait.StorageMap.html|
|✔️|[build-storage.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-storage.md)|**404**|``Storage Double Map``|https://docs.substrate.io/rustdocs/latest/frame_support/storage/trait.StorageDoubleMap.html|
|✔️|[build-storage.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-storage.md)|**404**|``Runtime Storage Page``|https://docs.substrate.io/v3/runtime/storage/|
|✔️|[build-smart-contracts.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-smart-contracts.md)|**404**|`contract pallet`|https://docs.substrate.io/rustdocs/latest/pallet_contracts/index.html|
|✔️|[build-smart-contracts.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-smart-contracts.md)|**404**|`Patract Labs`|https://patract.io/|
|✔️|[build-node-interaction.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-node-interaction.md)|**404**|`Substrate RPC API`|https://docs.substrate.io/rustdocs/latest/sc_rpc_api/index.html|
|⌛|[build-tools-index.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-tools-index.md)|**404**|`Tutorials`|https://dotleap.com/tag/tutorial/|
|✔️|[build-protocol-info.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-protocol-info.md)|**404**|``AccountData``|https://docs.substrate.io/rustdocs/latest/pallet_balances/struct.AccountData.html|
|✔️|[build-protocol-info.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/build/build-protocol-info.md)|**404**|`reference a previous transaction`|https://docs.substrate.io/rustdocs/latest/pallet_multisig/struct.Timepoint.html|
|⌛|[community.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/general/community.md)|**404**|`Dotleap.com Tutorials`|https://dotleap.com|
|✔️|[faq.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/general/faq.md)|**404**|`networking documentation`|https://docs.substrate.io/rustdocs/latest/sc_network/index.html|
|⌛||[kusama-community.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/general/kusama/kusama-community.md)|**404**|`Dotleap.com Tutorials`|https://dotleap.com|
|✔️|[learn-common-good-chains.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/learn/learn-common-good-chains.md)|**404**|`Statemint`|https://github.com/paritytech/cumulus/tree/master/polkadot-parachains/statemint|
|✔️|[learn-nft.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/learn/learn-nft.md)|**403**|`NFT on Ethereum`|https://opensea.io/assets/0x2127fe7ffce4380459cced92f2d4793f3af094a4/12598|
|✔️|[learn-treasury.md](https://github.com/w3f/polkadot-wiki/blob/master/docs/learn/learn-treasury.md)|**404**|`Docs`|https://docs.substrate.io/rustdocs/latest/pallet_treasury/index.html|

### CODE REVIEW

File: docs/build/build-protocol-info.md [hunk 1]
```diff
@@ -132,7 +132,7 @@ More info:
 
 - [Lockable Currency](https://paritytech.github.io/substrate/master/frame_support/traits/trait.LockableCurrency.html)
 - [Lock Withdraw Reasons](https://paritytech.github.io/substrate/master/frame_support/traits/struct.WithdrawReasons.html)
-- [Vesting Info](https://paritytech.github.io/substrate/master/pallet_vesting/struct.VestingInfo.html)
+- [Vesting](https://paritytech.github.io/substrate/master/pallet_vesting/index.html)
```
Comment (nuke-web3) [inline docs/build/build-protocol-info.md]: https://paritytech.github.io/substrate/master/pallet_vesting/struct.VestingInfo.html is correct and functional

File: docs/learn/learn-accounts.md [hunk 2]
```diff
@@ -446,7 +446,7 @@ Currently, the **DepositBase** equals `deposit(1, 88)` (key size is 32; value is
 bytes) and the **DepositFactor** equals `deposit(0, 32)` (additional address of 32 bytes).
 
 The deposit function in JavaScript is defined below, cribbed from the
-[Rust source](https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/src/constants.rs#L26).
+[Rust source](https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/constants/src/lib.rs#L30).
```
Comment (nuke-web3) [inline docs/learn/learn-accounts.md]: as the lines change, better to omit
```suggestion
[Rust source](https://github.com/paritytech/polkadot/blob/master/runtime/polkadot/constants/src/lib.rs).
```

File: docs/build/build-protocol-info.md (hunk 1 above)
Comment (alfarok) [reply docs/build/build-protocol-info.md]: Thanks for the review @NukeManDan!  I didn't see you were working on #3437 when I was working on this.  After pulling in your changes I noticed we had many updated urls that matched but I also had to resolved a few differences.  I must have overwrote this one accidentally with what I had. :+1:

File: docs/learn/learn-accounts.md (hunk 2 above)
Comment (alfarok) [reply docs/learn/learn-accounts.md]: agreed, I was updating what was existing but I guess we can just point to the `lib.rs` file here?

File: docs/build/build-protocol-info.md (hunk 1 above)
Comment (alfarok) [reply docs/build/build-protocol-info.md]: Also thanks for updating all the `docs.substrate.io` links!  That was next on my list but after your PR all the remaining urls appear valid and are no longer redirecting.

**Review by DrW3RK — APPROVED** [review]
Thanks for the fixes.

File: docs/learn/learn-accounts.md (hunk 2 above)
Comment (nuke-web3) [reply docs/learn/learn-accounts.md]: Yes, the context is important, actually evaluate if it makes sense to only reference the file, you may need to add " look for X in this source" to help hint at where it lives

**Pull request merged**